File: schemaorggo/employeerole.go
package schemaorggo

import "encoding/json"

// EmployeeRole see : https://schema.org/EmployeeRole
type EmployeeRole struct {
	OrganizationRole

	typeContext

	// BaseSalary see : https://schema.org/baseSalary
	// The base salary of the job or of an employee in an EmployeeRole.
	// types : MonetaryAmount Number PriceSpecification
	BaseSalary []interface{} `json:"baseSalary,omitempty"`

	// SalaryCurrency see : https://schema.org/salaryCurrency
	// The currency (coded using ISO 4217 (see: https://schema.orghttp://en.wikipedia.org/wiki/ISO_4217) ) used for the main salary information in this job posting or for this employee.
	// types : Text
	SalaryCurrency []string `json:"salaryCurrency,omitempty"`
}

func (v EmployeeRole) IntoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.OrganizationRole.IntoMap(intop)

	into := *intop

	{
		var value interface{} = v.BaseSalary
		if len(v.BaseSalary) == 1 {
			value = v.BaseSalary[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["baseSalary"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.SalaryCurrency
		if len(v.SalaryCurrency) == 1 {
			value = v.SalaryCurrency[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["salaryCurrency"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v EmployeeRole) AsMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.IntoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "EmployeeRole"

	return data, nil
}

func (v EmployeeRole) MarshalJSON() ([]byte, error) {
	data, err := v.AsMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/employee_role.php
<?php

class EmployeeRole extends OrganizationRole implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'EmployeeRole';
	
	/**
	 * The base salary of the job or of an employee in an EmployeeRole.
	 * see : https://schema.org/baseSalary
	 * @var \MonetaryAmount|\MonetaryAmount[]|float|float[]|\PriceSpecification|\PriceSpecification[]
	 */
	public var $base_salary;
	
	/**
	 * The currency (coded using ISO 4217 (see: https://schema.orghttp://en.wikipedia.org/wiki/ISO_4217) ) used for the main salary information in this job posting or for this employee.
	 * see : https://schema.org/salaryCurrency
	 * @var string|string[]
	 */
	public var $salary_currency;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'EmployeeRole'
		);
		
		$serialized = so_json_serialize( $this->base_salary );
		if ( ! empty( $serialized ) ) {
			$out['baseSalary'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->salary_currency );
		if ( ! empty( $serialized ) ) {
			$out['salaryCurrency'] = $serialized;
		}
		
		return $out;
	}
}
